pkg/geth/client/rpc_client.go

```diff
@@ -68,8 +78,20 @@ func (client RpcClient) BatchCall(batch []BatchElem) error {
 			Args:   batchElem.Args,
 			Error:  batchElem.Error,
 		}
-
 		rpcBatch = append(rpcBatch, newBatchElem)
 	}
 	return client.client.BatchCall(rpcBatch)
 }
+
+// Subscribe subscribes to an rpc "namespace_subscribe" subscription with the given channel
+// The first argument needs to be the method we wish to invoke
+func (client RpcClient) Subscribe(namespace string, payloadChan interface{}, args ...interface{}) (*rpc.ClientSubscription, error) {
+	chanVal := reflect.ValueOf(payloadChan)
+	if chanVal.Kind() != reflect.Chan || chanVal.Type().ChanDir()&reflect.SendDir == 0 {
+		return nil, errors.New("second argument to Subscribe must be a writable channel")
+	}
+	if chanVal.IsNil() {
```

Comment:
The first check is checking if the type is a kind of channel, more specifically a kind of channel we can send to, but there could still be a nil value for that channel type which is why the 2nd check is needed. Same pattern is used in geth [here](https://github.com/ethereum/go-ethereum/blob/master/rpc/client.go#L420)... which makes me realize this might be redundant 🤔